### DISCUSSION TITLE
ℹ️ About the Insomnia updates that will happen in the next few months.

### DISCUSSION THREAD

**Discussion by subnetmarco** [Kong/insomnia#6815] [Announcement]
Hi everyone 👋 You may have noticed that in the last few months we made some changes in the application that may introduce [unwanted](https://github.com/Kong/insomnia/issues/6803) [regressions](https://github.com/Kong/insomnia/issues/6719) in the usability of the product.

These regressions are happening because we are taking ownership of areas of the product that needed to be improved **dramatically**, and these changes may sometimes introduce unwanted effects because of the sheer complexity of the codebase that we are improving.

As such we have to make these changes to get better:

* The number of regressions will reduce dramatically in the coming months as we complete our work.
* We are very responsive to issues on this repository, and we would rather make a quick patch release to fix those unwanted regressions rather than being blocked.
* The performance and feature set of Insomnia will dramatically improve throughout 2024 as soon as we have completed this rough patch of work.

![insomnia_crane_2](https://github.com/Kong/insomnia/assets/964813/dc586fbf-618b-484d-b756-d209c218d2f0)

By unblocking this work, we will be able to work in short order on some of the most requested capabilities that you have asked:

* ~Introducing **pre-request scripts** into the application.~ (Shipped in 9.x)
* ~Introducing **per-request variables**.~ (Shipped in 8.6.x)
* Introducing support for **multiple tabs**.
* ~Being able to re-use authentications and environments across multiple collections. (Shipped in 9.x)~
* ~Introducing a vast framework for API testing. (Shipped in 9.x)~
* ~Work on performance improvements to accelerate the application by 50% across all areas. (Main performance improvements have shipped, with the bulk of improvements shipped in v9.3.3+)~
* ~Introducing global environments. (Shipped in 9.x)~
* ~Introducing path parameters. (Shipped in 9.x)~
* And more.

I wanted to thank everyone for the patience, with a strong commitment to making all of this better. Your issues on GitHub are also important to identify these regressions, so please keep them coming and we will include them in the next patch releases.

**Comment by subnetmarco** [comment]
We just shipped v8.4.4 with many performance improvements compared to v8.4.2 🏎 and we fixed some long lasting bugs like https://github.com/Kong/insomnia/issues/1937 and https://github.com/Kong/insomnia/issues/6387 among those.

**Comment by 64J0** [comment]
My local configuration doesn't appear anymore after the update...

**Reply by 64J0** [reply]
After updating:

![image](https://github.com/Kong/insomnia/assets/50725287/89cb895d-7b1c-4f5c-aee1-b2b9f6a318aa)

**Reply by subnetmarco** [reply]
What version did you update from?

**Reply by 64J0** [reply]
I was using `8.4.4`, and now I'm using `8.4.5`

**Reply by 64J0** [reply]
It was not required to authenticate in the version I was using

**Reply by subnetmarco** [reply]
Then you must have updated from < 8.x - Could that have been the case?

**Reply by 64J0** [reply]
No, I was using 8.x for a while.

**Reply by 64J0** [reply]
I discovered the problem now. It was required to manually move the data in this panel:

![image](https://github.com/Kong/insomnia/assets/50725287/c865cf87-a16d-4f0a-a3c6-46a273e4b449)

After selecting the organization and hitting the "Move" button, it appeared again. But this is something that must be better documented, or automatically loaded. It was really bad to "lose" my configuration after the update.

**Reply by gatzjames** [reply]
Hi @64J0.
Could it be that you were you logged in with another account before the update?
Having your projects appear in the untracked projects means that they were under a different organization before that your current account doesn't have access to.

**Reply by 64J0** [reply]
That could be the case, but it would be pretty weird. As far as I know I have only 1 account.

**Comment by relgames** [comment]
Hi @subnetmarco , I don't believe this statement is true:
> We are very responsive to issues on this repository, and we would rather make a quick patch release to fix those unwanted regressions rather than being blocked.

For example, this regression is open since August 16th https://github.com/Kong/insomnia/issues/6301 and there is still no fix. 
This one https://github.com/Kong/insomnia/issues/5851 is open since March and also no fix yet.

**Reply by subnetmarco** [reply]
Hi @relgames, as usual it's a game of prioritization. Let's take a look at those two, and see if we can fix them in our upcoming December release.

**Reply by subnetmarco** [reply]
We couldn't make it in December, but these two issues are prioritized for the January release.

**Reply by subnetmarco** [reply]
These two issues have been resolved and it is currently available in [8.5.2 beta](https://github.com/Kong/insomnia/releases/), it will be GA at the next GA release (v8.5.2).

**Comment by subnetmarco** [comment]
We have shipped [8.5.x](https://github.com/Kong/insomnia/releases) that fixes dated issues, notably we have rewritten the "Environment" section which should behave a lot better and we have fixed issues like #2255 and we have added a new footer section to notify the user of untracked projects that may appear to be missing, but are still on the computer but not associated to an organization:

<img width="432" alt="Screenshot 2023-12-21 at 1 34 31 PM" src="https://github.com/Kong/insomnia/assets/964813/7b8a56cb-451e-46f8-aacf-7e1a22558d86">


# Next up 

Jan-March will be a very busy time for Insomnia:

* Iterating the top 10 requests in the issues and closing them.
* Pre-request scripting.
* Native API mocking.
* Start working on multiple-tabs.
* Making E2EE passphrases optional (but still available if the user wants them).
* And a lot more.

Happy holidays 🎄

**Reply by subnetmarco** [reply]
The command palette will be back in January: https://github.com/Kong/insomnia/issues/6301

**Comment by subnetmarco** [comment]
A new document finder is shipping in the next version of Insomnia (**CMD+P**). It is available today in 8.5.2 beta and it works from within a collection, before GA we are planning to make it work anywhere and also will be able to search across all organizations.

Try it out:

![image](https://github.com/Kong/insomnia/assets/964813/e2678d62-0194-459a-94c6-f11c944d56d6)

**Comment by subnetmarco** [comment]
We just shipped path parameters and this week we will also simplify how to value environment variables when it's missing.

In the meantime we are working on adding support for **pre-request scripts**.

More info at: https://github.com/Kong/insomnia/discussions/3551#discussioncomment-8176005

![image](https://github.com/Kong/insomnia/assets/964813/ee401b0c-43a9-4fd8-9dc5-e09a407e148b)

**Reply by DusanLesan** [reply]
@subnetmarco Path parameters are a cool feature. Now that I know about it, how do I dismiss `Path parameters are url path segments...` hint so it is not shown under the requests that do not need that?

**Reply by subnetmarco** [reply]
We will add a dismiss button.

**Reply by DusanLesan** [reply]
Great, thanks. I will use this meanwhile: 
```
[class^="text-sm"] {
    visibility: hidden;
}
```

**Reply by subnetmarco** [reply]
@CurryYangxx for visibility on the dismiss button for this notification, as discussed.

**Reply by CurryYangxx** [reply]
Got it!

**Reply by CurryYangxx** [reply]
The dismiss button will be added in this PR: https://github.com/Kong/insomnia/pull/7271

**Comment by subnetmarco** [comment]
The latest and greatest about pre-request scripts can be found in this [video demonstration](https://github.com/Kong/insomnia/discussions/3806#discussioncomment-8321782).

**Comment by subnetmarco** [comment]
🔄 We are making significant improvements to the synchronization experience: https://github.com/Kong/insomnia/pull/7098

**Comment by subnetmarco** [comment]
Don't miss pre-request scripting coming in the next release, the [beta is available now](https://github.com/Kong/insomnia/discussions/3806#discussioncomment-8725886).

**Comment by subnetmarco** [comment]
We have fixed all the shortcut problems in the upcoming Insomnia 9.0, for which there is a beta available now in the releases.

**Comment by ThaDaVos** [comment]
Sorry to comment/mention this, but when can we expect an update on the following idea/feature-request - as others, I expected it to function already when gRPC support was implemented:
https://github.com/Kong/insomnia/discussions/6358

**Reply by subnetmarco** [reply]
We can take a look at this after 9.0 GA is out next week.

**Comment by subnetmarco** [comment]
Insomnia 9.0 GA has been [released today](https://konghq.com/blog/product-releases/insomnia-9-0-pre-request-scripting-api-mocking) with pre-request scripting, improved offline collaboration, native API mocking and more.

**Reply by DusanLesan** [reply]
I have updated to 9.0.0, and I immediately have two complaints:

- Request list now have a minimal width set too wide - I cannot shrink it as far as I want. It is double the width I need, and I want that space for other two sections. If I attempt to shrink, the whole element gets hidden which is not what I want
- The line number in response preview takes constant size that would be enough for like 7+ digit numbers but my responses rarely need that much. My expectation is for line number to take only space needed. If response is only triple digit - shrink the line count element. 

I can create the issues if needed for better tracking. I have complaint about old behavior as well: I want last view to be opened on launch: If i was in a collection, I want to be brought there immediately after launch

**Reply by subnetmarco** [reply]
Having those issues would be great for tracking, indeed 🙏

**Reply by DusanLesan** [reply]
I have created https://github.com/Kong/insomnia/issues/7323 and https://github.com/Kong/insomnia/issues/7324 for these issues and found that I have already written the issue for last view: https://github.com/Kong/insomnia/issues/6928

**Reply by subnetmarco** [reply]
@DusanLesan One of those issues has been fixed in 9.1, the other one still stands.

**Reply by DusanLesan** [reply]
I have seen 7324 already in the latest update, thanks. The other one might not be so easy since there are some concerns raised there.

**Reply by DusanLesan** [reply]
I have created a new discussion detailing proposed improvement for [#7324](https://github.com/Kong/insomnia/issues/7324): [feature request](https://github.com/Kong/insomnia/discussions/7424)

**Comment by subnetmarco** [comment]
Insomnia 9.2 [has been released](https://konghq.com/blog/product-releases/announcing-insomnia-9-2) with support for 100+ randomly generated data types, global search palette across multiple organizations, and performance improvements when working with large OpenAPI files.

**Comment by subnetmarco** [comment]
Post request scripting is available in Insomnia 9.3 beta: https://github.com/Kong/insomnia/discussions/6242#discussioncomment-9533769

**Reply by smiller0924** [reply]
How do you download the beta version 9.3 or Insomnia?

**Reply by CurryYangxx** [reply]
> How do you download the beta version 9.3 or Insomnia?

Hi @smiller0924 You could download here: https://github.com/Kong/insomnia/releases

**Reply by subnetmarco** [reply]
And it's now available GA: https://github.com/Kong/insomnia/discussions/6815#discussioncomment-9895341

**Comment by subnetmarco** [comment]
Insomnia 9.3 [has been released](https://konghq.com/blog/product-releases/insomnia-9-3-ga) with after-response scripting, global environments, folder-level configuration, and more.

**Reply by DusanLesan** [reply]
I have two issues with this one:

- The environment indicator is replaced with a "Manage Environments" option, forcing me to click on it to see the current selection.
- The pane scroll bars have become ugly for some reason. I have tried several dark themes, and all now have light gray bars.

I would also like to bring attention to this ignored issue that makes using this app a terrible experience for my whole team: https://github.com/Kong/insomnia/issues/6070

**Reply by subnetmarco** [reply]
> The environment indicator is replaced with a "Manage Environments" option, forcing me to click on it to see the current selection.

This [has been fixed already](https://github.com/Kong/insomnia/issues/7615#issuecomment-2199622335) and will ship in the next minor.

> The pane scroll bars have become ugly for some reason. I have tried several dark themes, and all now have light gray bars.

@gatzjames for visibility.

**Reply by DusanLesan** [reply]
@gatzjames Do you need a issue ticket for the scroll bar issue?

**Comment by subnetmarco** [comment]
**Test Results** and **Collection Runner** coming soon, below a screenshot of the new Test Results section. If you have feedback on these capabilities, this is a good time to bring it up.

![image](https://github.com/Kong/insomnia/assets/964813/2ab7005d-4b3f-411a-a532-cf9e168a9860)

**Reply by DusanLesan** [reply]
I would appreciate it if you could prioritize the stability of core functionality over adding new features. For instance, it would be nice to send requests without encountering issues like: https://github.com/Kong/insomnia/issues/7693 and https://github.com/Kong/insomnia/issues/7696.

**Reply by subnetmarco** [reply]
@DusanLesan we are addressing both angles, including a new minor version of Insomnia that will solve for many of these stability issues, including the ones that you have reported.

**Reply by DusanLesan** [reply]
Sounds good. However, based on my recent experience with this app, I feel that its stability might not be receiving sufficient attention. I would like to advocate for a greater focus on stability than it currently has.

**Reply by DusanLesan** [reply]
Given that I am used to working with applications that mostly do not introduce issues that can be found within 10 minutes of testing, my perception might be biased towards app stability and clean UI. Have you considered having a poll discussion with options like "more focus on new features" and "more focus on app stability"?

**Reply by DusanLesan** [reply]
I forgot the third option since that one sounds wrong to me: "keep it going as it is"

**Reply by subnetmarco** [reply]
@DusanLesan Try to download the latest beta, many improvements have been built and will ship next week: https://github.com/Kong/insomnia/releases/tag/core%409.3.3-beta.3

**Reply by DusanLesan** [reply]
I have quickly tried it.

- Remembering the last position now works.
- Scroll bars look good again (even though hovering over request list bar is showing me horizontal resize cursor).
- Clicking on environment variable boxes works after the first press.
- Changing variables works fine now.
- Changing parameters before sending a request works again.
- I can open folders in the request list without aiming for the icon (even drag and move works).

Not really a bug, but a small complaint about the UX for moving requests below the expanded folder is that it will be placed at the level of the last item it was moved over. That was not super obvious to me, but I do not mind it now that I know how it works (probably only an issue when attempting to move a request after an expanded folder at the list's end).

All in all, I'm happy with the fixes and hope there are no big bugs that will pop up at the worst time.

**Comment by IGLOU-EU** [comment]
Hi Team,

I am experiencing significant issues with the latest update of Insomnia. I can no longer work locally without interruptions. It appears I need an account with you to proceed, and I am unable to access my previous projects unless I associate them with an organization. When I try to do so, I am prompted to connect to your cloud service.

Is there a way to work seamless local work without these complications? It feels as though the transition to a closed ecosystem is limiting the freedom that open-source software typically provides.

Can you please clarify whether this is an intentional move towards a full SaaS model, or if this is a mistake that will be rectified?

Thank you for your attention to this matter.

**Reply by subnetmarco** [reply]
Hi @IGLOU-EU, you can still store all of your data locally if you desire to do so by setting the storage configuration in the project settings:

![Screenshot 2024-07-26 at 9 16 04 AM](https://github.com/user-attachments/assets/f6d12b0d-7cfd-40de-8d4a-d07260091635)

Approximately a year ago we decided to introduce signups for Insomnia users to bring the project back on a sustainable path and grow it into a business (which has been established, as you can see by the many improvements over the last 12 months), so whether you decide to use the cloud collaboration or retain all your projects locally, you still need to create a free account.

Other than that nothing changes: the capabilities are free, open, and Insomnia is definitely not turning into a closed ecosystem. We also made collaboration free for everyone in the free plan if you have one project, so it's more free than it used to be in the past (in the past cloud collaboration was a premium feature).

Requiring an account has caused some pushback in the past because of the mandatory login (unless you use Scratch Pad, which can be used locally without a login but it is more limited in capabilities). But we remember the login across application restarts, and you only need to do it once, and it's free. We also introduced capabilities - like API Mocking - that require an account to be delivered and will continue to introduce more.

I hope this answers your questions.

**Reply by IGLOU-EU** [reply]
I know, it's free, but who want to create an account for every software ? Sell Free software is ok, no problem with that. 
But forcing to create an account ... that hard. For now, I leave, I will be testing firecamp, hoppscotch and bruno. It probably not require to sign-in

:disappointed:

**Reply by subnetmarco** [reply]
We require a free account because overall it allows us to finance the continued development of Insomnia with the subset of users that decide to convert to a premium plan. This allows us to build a continuous amount of value that is being delivered to you for free, even capabilities that - in the other products you have mentioned - are currently offered behind a paywall but are still free in Insomnia.

For many Insomnia users - and for us - this ends up being a good "contract" since it costs you nothing and creates an enormous amount of product value in exchange. But of course I accept that it may not be acceptable for everybody, we simply can't make everybody happy, and I hope we can win you back someday.

**Reply by DusanLesan** [reply]
@subnetmarco Could you please explain what value you derive from the accounts of people using free plans? How exactly is my data being used?

**Reply by subnetmarco** [reply]
It allows us to grow and build the product.

The data sits on our servers, and it introduces the user to capabilities (like collaboration, mocking, etc) pretty much immediately in the journey of the product. In other words, users **with** an account have a higher conversion rate in using the premium features, than users **without** an account. Even free users that do not use these capabilities immediately, eventually will be more likely to use them with an account (but of course nobody is forced to do it).

This allows us to hire more people in the team, build a bigger product, introducing new capabilities, supporting the community and overall have the ability to execute on our vision for Insomnia.

Open source software has changed a lot over the past few years, because modern OSS is very complex and requires a lot of resources, it may be free to use but it's not cheap to build. Especially for an application like Insomnia that is being used by millions of developers. It's not fair for open source software (not specifically Insomnia, but in general) to both be pushed to get better and better over time, but at the same time ignoring the dynamics that allow the maintainers to keep investing into the product in a sustainable way.

I don't believe that Insomnia should gate keep critical API debugging and testing capabilities behind a paywall either, because I am a proponent that the core capabilities should be free. So instead of removing GraphQL, WebSockets and something else behind a yearly or monthly fee, I would rather offer all the core capabilities for free in exchange for a 1-click free signup because I believe that - by doing it this way - we can attract a larger number of people to use our product that we are spending so much time, effort and dedication to build and we can therefore create a bigger impact.

**Reply by DusanLesan** [reply]
Ok, thank you for explaining.

**Reply by IGLOU-EU** [reply]
Hi @subnetmarco ,

As I mentioned earlier, making money from a FOSS project isn't bad, and I understand your rationale. The principle of "free as in free speech, not as in free beer." is clear to me.

However, I must voice my criticism transparently. I don't appreciate being forced to sign up for a cloud service to use local software, and I especially dislike the intentional restrictions placed on the software to drive this.

I know I'm likely not alone in feeling this way, though perhaps I'm one of the few who are vocal about it.

**Comment by subnetmarco** [comment]
Insomnia 9.3.3 GA [has been released](https://konghq.com/blog/product-releases/insomnia-9-3-3) with up to 20x performance improvements, and many more enhancements including a refactored sidebar that solves many problems we experienced in the previous versions.

**Comment by subnetmarco** [comment]
Stay tuned for the new free and unlimited **Collection Runner** which is shipping in Insomnia v10 in September:

https://github.com/user-attachments/assets/fc2ef325-e91d-4b58-a948-922ac74cf270
